File: backend/app/models/sale.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Float, Integer, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base


class Sale(Base):
    __tablename__ = "sales"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    product_id = Column(String, ForeignKey("inventory.id"), nullable=False, index=True)
    quantity = Column(Integer, nullable=False)
    unit_price = Column(Float, nullable=False)
    total = Column(Float, nullable=False)
    customer = Column(String(255), default="")
    user_id = Column(String, ForeignKey("users.id"), nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    product = relationship("InventoryItem", back_populates="sales")

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/Sale.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("sales")]
public class Sale : BaseEntity
{
    [Column("product_id")]
    public string ProductId { get; set; } = string.Empty;

    [Column("quantity")]
    public decimal Quantity { get; set; }

    [Column("unit_price")]
    public decimal UnitPrice { get; set; }

    [Column("total")]
    public decimal Total { get; set; }

    [Column("customer")]
    public string? Customer { get; set; }

    [Column("user_id")]
    public string? UserId { get; set; }

    [JsonIgnore]
    public InventoryItem Product { get; set; } = null!;
}
